server/routes/index.js

```diff
@@ -1,7 +1,17 @@
 const router = require('express').Router();
 
+const {signup} = require('../controllers');
+const { getProductByCId } = require('../controllers/getProductsByCId');
 const { getProducts, getProductData } = require('../controllers/products');
 
+
+router.post('/register', signup);
+
+router.get('categories/', getProductByCId)
```

Comment:
You can set the `userId` as a param from here or whatever is `cid` 
```suggestion
router.get('categories/:productId', getProductByCId)
```